test/link-check/linkCompare.test.js

```diff
@@ -0,0 +1,482 @@
+
+      expect(result.match).to.be.true;
+      expect(result.unique).to.have.lengthOf(0);
+      expect(result.links).to.have.lengthOf(5);
+    });
+
+    it('Compares links from different Markdown files', async () => {
+      const source1 = `${pathname}mock/links1.md`;
+      const source2 = `${pathname}mock/links2.md`;
+      const result = await compare(source1, source2);
+
+      expect(result.match).to.be.false;
+      expect(result.unique).to.have.lengthOf(4);
+      expect(result.links).to.deep.equal([
+        { index: 0, link1: linkURLs[0], link2: linkURLs[4], match: false },
+        { index: 1, link1: linkURLs[1], link2: linkURLs[3], match: false },
+        { index: 2, link1: linkURLs[2], link2: linkURLs[2], match: true },
+        { index: 3, link1: linkURLs[3], link2: linkURLs[1], match: false },
+        { index: 4, link1: linkURLs[4], link2: linkURLs[0], match: false },
+      ]);
+    });
+
+    it('Compares links from Markdown and HTML files', async () => {
+      const source1 = `${pathname}mock/links1.md`;
+      const source2 = 'https://business.adobe.com/links1.html';
+      const html = fs.readFileSync(`${pathname}mock/links1.html`, 'utf8');
+      const stubFetch = stub().resolves({ ok: true, text: stub().resolves(html) });
+      const result = await compare(source1, source2, stubFetch);
+
+      expect(result.match).to.be.true;
+      expect(result.unique).to.have.lengthOf(0);
+      expect(result.links).to.have.lengthOf(5);
+    });
+
+    it('Compares links from Markdown and Docx files', async () => {
+      const source1 = `${pathname}mock/links1.md`;
+      const source2 = `${pathname}mock/links1.docx`;
+      const result = await compare(source1, source2);
+
+      expect(result.match).to.be.true;
+      expect(result.unique).to.have.lengthOf(0);
+      expect(result.links).to.have.lengthOf(5);
+    });
+
+    it('Compares links from Docx and HTML files', async () => {
+      const source1 = `${pathname}mock/links2.docx`;
+      const source2 = 'https://business.adobe.com/links2.html';
+      const html = fs.readFileSync(`${pathname}mock/links2.html`, 'utf8');
+      const stubFetch = stub().resolves({ ok: true, text: stub().resolves(html) });
+      const result = await compare(source1, source2, stubFetch);
+
+      expect(result.match).to.be.true;
+      expect(result.unique).to.have.lengthOf(0);
+      expect(result.links).to.have.lengthOf(5);
+    });
+  });
+
+  describe('Compares real files', () => {
+    it('omnichannel-retail-trends business website to bacom', async () => {
+      const source1 = `${pathname}mock/omnichannel-retail-trends-business.md`;
+      const source2 = `${pathname}mock/omnichannel-retail-trends-bacom.md`;
+      const result = await compare(source1, source2);
+
+      expect(result.match).to.be.false;
+      expect(result.unique).to.have.lengthOf(29);
+      expect(result.links).to.have.lengthOf(36);
+      expect(result.unique).to.deep.equal([
```

Comment:
could store the below deep.equal `[]` in its own file? Same for 394?

Reply:
Good idea